protocol/messenger_messages_order_controller_test.go

```diff
@@ -0,0 +1,126 @@
+package protocol
+
+import (
+	"bytes"
+	"sort"
+	"sync"
+
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/protocol/transport"
+)
+
+type messagesOrderType int
+
+const (
+	messagesOrderRandom messagesOrderType = iota
+	messagesOrderInPostOrder
+	messagesOrderReversed
```

Comment:
future idea: we could also implement `OrderManual`
So that the user could explicitly set the order of messages on the fly. But maybe it makes no sense, because we could just change the order of posting 🙂

Reply:
Indeed that would be ideal, see my comment in tests: https://github.com/status-im/status-go/blob/8d8e60f5edda12a5be1e684470e3daaea983579c/protocol/communities_events_eventual_consistency_test.go#L128

This will not be straightforward to implement, though, as we would need to know the message hashes given API triggers. Also, some messages are sent asynchronously (e.g., CommunityDescription), making it even more difficult to figure out.